src/analysis/vector/geometry_checker/qgsgeometrycheckcontext.h

```diff
@@ -58,6 +59,15 @@ struct ANALYSIS_EXPORT QgsGeometryCheckContext
      */
     const QgsCoordinateTransformContext transformContext;
 
+    /**
+     * The project can be used to resolve additional layers.
+     *
+     * This must only be accessed from the main thread (i.e. do not access from the collectError method)
+     *
+     * \since QGIS 3.10
+     */
+    const QgsProject *project;
```

Comment:
Might be safest to make this a function, and Q_ASSERT that it IS being called only from the main thread